chia/data_layer/download_data.py

```diff
@@ -149,53 +231,56 @@ async def insert_from_delta_file(
     log: logging.Logger,
     proxy_url: str,
     downloader: Optional[PluginRemote],
+    group_files_by_store: bool = False,
 ) -> bool:
+    if group_files_by_store:
+        client_foldername.joinpath(f"{store_id}").mkdir(parents=True, exist_ok=True)
+
     for root_hash in root_hashes:
         timestamp = int(time.time())
         existing_generation += 1
-        filename = get_delta_filename(store_id, root_hash, existing_generation)
-        request_json = {"url": server_info.url, "client_folder": str(client_foldername), "filename": filename}
-        target_path = client_foldername.joinpath(filename)
-        filename_exists = False
-        if target_path.exists():
-            filename_exists = True
-            log.info(f"Filename {filename} exists, don't download it.")
-        else:
-            if downloader is None:
-                # use http downloader - this raises on any error
-                try:
-                    await http_download(client_foldername, filename, proxy_url, server_info, timeout, log)
-                except (asyncio.TimeoutError, aiohttp.ClientError):
-                    new_server_info = await data_store.server_misses_file(store_id, server_info, timestamp)
-                    log.info(
-                        f"Failed to download {filename} from {new_server_info.url}."
-                        f"Miss {new_server_info.num_consecutive_failures}."
-                    )
-                    log.info(f"Next attempt from {new_server_info.url} in {new_server_info.ignore_till - timestamp}s.")
-                    return False
-            else:
-                log.info(f"Using downloader {downloader} for store {store_id.hex()}.")
-                async with aiohttp.ClientSession() as session:
-                    async with session.post(
-                        downloader.url + "/download",
-                        json=request_json,
-                        headers=downloader.headers,
-                    ) as response:
-                        res_json = await response.json()
-                        if not res_json["downloaded"]:
-                            log.error(f"Failed to download delta file {filename} from {downloader}: {res_json}")
-                            break
-
-        log.info(f"Successfully downloaded delta file {filename}.")
+        target_filename_path = get_delta_filename_path(
+            client_foldername, store_id, root_hash, existing_generation, group_files_by_store
+        )
+        filename_exists = target_filename_path.exists()
+        attempts = 0
+        for grouped_by_store in (False, True):
+            attempts += 1
```

Comment:
Or drop the `, 1` and make this zero-based counting.

```suggestion
        for attempts, grouped_by_store in enumerate((False, True), 1):
```

Reply:
Or perhaps this really is a case where an `else:` on the `for` is cleanest.

```python
for x in stuff:
    if is_what_we_want(x):
        break
else:
    raise Exception("did not find what we wanted")
```